This repository contains annotated scripts used to analyse CUT&RUN data performed targeting SMAD4 in M170117 melanoma cells. 


Excerpt from Materials and Methods:
CUT&RUN Low Volume Urea
M170117 cells were cultured in at 37°C, 5% CO2, and 95% humidity in RPMI medium (11875093, Thermo Scientific) supplemented with 10% Calf Bovine Serum (12133C, Sigma-Aldrich), 10 U/ml Penicillin-Streptomycin (15276355, Gibco) and 4 mM L-Glutamine (11500626, Gibco). Passages were performed with 2 mM EDTA-PBS (AM9260G, Thermo Fisher Scientific). Stimulations were performed with TGFβ1 (RnD Systems 7754-BH-005, Stock: 10 ug/mL in 4mM HCl + 0.1% BSA, 1:5000 dilution of stock), and/or MEKi (Trametinib, Stock: 50 uM Active Biochemicals - A-1258, 1:1000 dilution of stock), and/or DMSO 20 hours prior to harvest for CUT&RUN. Cells were detached with 2 mM EDTA-PBS for 5 min and then centrifuged 250 g for 5 min. Cells were resuspended in 2 ml PBS and counted; 250,000 cells were taken per replicate (N = 2 per condition). CUT&RUN Low Volume Urea, library preparation and sequencing were performed as described in (https://doi.org/10.1242/dev.201124), using anti-SMAD4 (ABIN6972727, antibodies-online), anti-SMAD4 (D3R4N #46535, Cell Signaling Technology), and IgG negative control (ABIN101961, antibodies-online). Samples were sequenced pair-end to a depth of 5 – 15 million reads. 

CUT&RUN Data Analysis
Reads were trimmed with bbmap bbduk (https://doi.org/10.1371/journal.pone.0185056, version 38.18) to remove adapter sequences, known artifacts, and poly [AT], G and C repeats. Reads were aligned to the hg38 genome with bowtie2 (doi:10.1038/nmeth.1923, version 2.4.5) with the options –local –very-sensitive-local –no-unal –no-mixed -no-discordant –phred33 –dovetail -I 0 -X 500. Samtools (https://doi.org/10.1093/bioinformatics/btp352, version 1.11) suite was used to remove duplicate and incorrectly paired reads. Bedtools (https://doi.org/10.1093/bioinformatics/btq033, version 2.30.0) was used to remove reads mapped to the CUT&RUN hg38 blacklist (https://doi.org/10.1101/2022.11.11.516118) from bam files. Individual track bedgraphs were created using bedtools genomecov on pair-end mode. For visualization, bam files of biological replicates were merged using samtools and then signal per million reads tracks were created by using the --SPRM function of macs2 (https://doi.org/10.1186/gb-2008-9-9-r137, version 2.2.6) for each replicate pool with the options -f BAMPE --keep-dup all --SPMR and –bdg. Peaks were individually called using macs2 for each replicate against the negative control using the options -f BAMPE –keep-dup-all and -p 1e-2. Output narrowPeak files were overlapped using Intervene (https://doi.org/10.1186/s12859-017-1708-7, version 0.6.4), keeping only reproducible peak regions called in both replicates. Venn diagrams were created using Intervene. Motif analysis was done using HOMER (https://doi.org/10.1016/j.molcel.2010.05.004, version 4.11) findMotifsGenome to find motifs in the hg38 genome using -size given, and peak region annotation was done with HOMER annotatePeaks on default settings. Peak set gene annotation was done using GREAT (10.1038/nbt.1630, version 4.0.4) with default parameters. Signal intensity plots were created using ngsplot (https://doi.org/10.1186/1471-2164-15-284, version 2.63) with options -N 4 -GO none -SC global. Enrichr (10.1186/1471-2105-14-128) was used for pathway analysis with the Bioplanet 2019 pathways. 
